File: android/src/main/java/com/iconshot/detonator/request/AudioSeekRequest.java
package com.iconshot.detonator.request;

import androidx.media3.common.Player;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.element.AudioElement;
import com.iconshot.detonator.tree.Edge;

public class AudioSeekRequest extends Request<Integer> {
    public AudioSeekRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    public void run() {
        Integer position = decode(Integer.class);

        Edge edge = getComponentEdge();

        Edge audioEdge = edge.children.get(0);

        AudioElement element = (AudioElement) audioEdge.element;

        Player player = (Player) element.player;

        if (player != null) {
            player.seekTo(position);
        }

        end();
    }
}

File: ios/Sources/Detonator/Request/AudioSeekRequest.swift
import AVFoundation

class AudioSeekRequest: Request {
    override public func run() {
        let position: Int! = decodeData()
        
        let edge = getComponentEdge()!
        
        let audioEdge = edge.children[0]
        
        let element = audioEdge.element as! AudioElement
        
        let seconds = Double(position) / 1000
    
        let time = CMTime(seconds: seconds, preferredTimescale: 600)
        
        element.player?.seek(to: time)
        
        end()
    }
}
